### ISSUE TITLE
LATEST considers a bucket whose start time doesn't fall within [fromTimestamp, toTimestamp]

### ISSUE THREAD

**Issue by OwsOwies** [RedisTimeSeries/RedisTimeSeries#1486]
Hi, having an issue with `LATEST`. Having an hourly compaction series with the following data:
```
$ redis-cli TS.RANGE key - + LATEST
1) 1) (integer) 1671087600000
   2) 18500000000
2) 1) (integer) 1671091200000
   2) 7700000000
3) 1) (integer) 1671624000000
   2) 700000000
4) 1) (integer) 1671627600000
   2) 5900000000
5) 1) (integer) 1672650000000
   2) 60700000000
6) 1) (integer) 1672653600000
   2) 106500000000
7) 1) (integer) 1673967600000 // This is not here without LATEST
   2) 2400000000
```
Making the following query includes the latest compacted value whose start time doesn't fall within `[fromTimestamp, toTimestamp]`.
```
$ redis-cli TS.RANGE key 1685570400000 1688162399999 LATEST
1) 1) (integer) 1673967600000
   2) 2400000000
```
From: Jun 01 2023
To: Jun 30 2023
Returned: Jan 17 2023

What's wrong?

On a side note I'd also like a clarification on what `ALIGN` does:
```
$ redis-cli TS.RANGE key fromTimestamp toTimestamp LATEST AGGREGATION SUM {toTimestamp - fromTimestamp} ALIGN fromTimestamp
```
Would this query correctly aggregate **ONLY** values of timestamps between fromTimestamp to toTimestamp into a single bucket considering **ALL** points which fall in between from and to? The documentation on ALIGN is a bit ambiguous to me.
As I understand, without ALIGN keyword, there could be multiple buckets, not really falling in between from and to, so I would like to clarify that ALIGN serves my use case.

Thanks!

**Comment by LiorKogan** [comment]
This seems like a bug. We will try to reproduce it.
If you can provide a minimal, complete example - it would help.

Which version or RedisTimeSeries are you using?

Regarding `ALIGN`:
Your assumptions are correct.
By default, aggregation buckets of one hour will contain data for [00:00 .. 01:00), [01:00 .. 02:00) etc.
But suppose you want the buckets to be [00:30 .. 01:30), [01:30 .. 02:30) etc. - you can align the buckets to 00:30 after the Epoch (or on any other date). That way, you force one aggregation bucket to start at the specified time, and all other buckets are aligned accordingly.

`AGGREGATION SUM {toTimestamp - fromTimestamp} ALIGN fromTimestamp` should indeed create a single bucket for the period between fromTimestamp and toTimestamp. Maybe you should use `AGGREGATION SUM {toTimestamp - fromTimestamp + 1} ALIGN fromTimestamp`, please try this if you receive two buckets.

**Comment by LiorKogan** [comment]
Fixed in RedisTimeSeries 1.8.11
